src/taskEvents/utils.js

```diff
@@ -107,6 +111,97 @@ async function getData(data) {
   };
 }
 
+async function getUserSkills(id){
+  try{
+    const res = await fetch(`${BASE_URL}/users/${id}/skills`, {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Content-type': 'application/json',
+      }
+    });
+    
+    const userSkills = await res.json();
+    return userSkills;
+  } catch (error) {
+    console.log(error)
+  }
+}
+
+async function getTagLevelOptions() {
+  try{
+    const levelsResponse = await fetch(`${BASE_URL}/levels`, {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Content-type': 'application/json',
+      }
+    });
+    const tagsResponse = await fetch(`${BASE_URL}/tags`, {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Content-type': 'application/json',
+      }
+    });
+    
+    const { allTags } = await tagsResponse.json();
+    let { allLevels } = await levelsResponse.json();
+    allLevels = allLevels.sort((a,b) => {
+      if(parseInt(a.name) < parseInt(b.name)) return -1;
+      if(parseInt(a.name) > parseInt(b.name)) return 1;
+      return 0
+  }) 
+    return { allLevels, allTags };
+  } catch (error) {
+    console.log(error)
+  }
+}
+
+async function addSkillToUser(skillToAdd, levelToAdd, userid) {
+  const body = {
+    itemid: userid, 
+    itemType: "USER",
+    tagPayload: [{
+        levelid: levelToAdd?.id,
+        tagid: skillToAdd?.id
+    }]
+  }
+  try{
+    const response = await fetch(`${BASE_URL}/items`,{
+      method: 'POST',
+      credentials: 'include',
+      body: JSON.stringify(body),
+      headers: {
+        'Content-type': 'application/json',
+      } 
+    })
+    await response.json()
```

Comment:
We should check for `response.ok` value before this because in case of 404 it will not raise an error
Reference https://dmitripavlutin.com/javascript-fetch-async-await/#3-handling-fetch-errors

Reply:
done thanks for sharing the ref.